pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,8 +120,70 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .filter(t -> {
+                               if (t.getInserted().contains("/")) {
+                                   return xpathExpressionArea.getText().substring(xpathExpressionArea.getText().lastIndexOf("/", xpathExpressionArea.getText()
+                                                                               .lastIndexOf("/"))).matches("[a-zA-Z/]");
+                               } else {
+                                   return xpathExpressionArea.getText().substring(xpathExpressionArea
+                                                                                               .getText().lastIndexOf(("/"))).matches("[a-zA-Z/]");
+                               }
+                           })
+                           .subscribe(t -> {
+                               if (t.getInserted().equals("/")) {
+                                   autoComplete(xpathExpressionArea.getText().substring(xpathExpressionArea
+                                       .getText().lastIndexOf("/", xpathExpressionArea.getText().lastIndexOf("/"))));
+                               } else {
+                                   autoComplete(xpathExpressionArea.getText().substring(xpathExpressionArea
+                                                                                                    .getText().lastIndexOf(("/"))));
+                               }
+
+                           });
     }
 
+    private void autoComplete(String input) {
+
+        autoCompletePopup.getItems().clear();
+        List<MenuItem> resultToDisplay;
+
+        XPathSuggestions xPathSuggestions = new XPathSuggestions(parent.getLanguageVersion().getShortName());
```

Comment:
A previous comment of mine mentioned passing a Language instance to the constructor and you moved further away from that